libs/dal/src/repositories/tenant/tenant.repository.ts

```diff
@@ -3,8 +3,9 @@ import { SoftDeleteModel } from 'mongoose-delete';
 import { BaseRepository } from '../base-repository';
 import { TenantDBModel, TenantEntity } from './tenant.entity';
 import { Tenant } from './tenant.schema';
+import { EnforceEnvId, EnforceEnvOrOrgIds } from '../../types';
 
-export class TenantRepository extends BaseRepository<TenantDBModel, TenantEntity> {
+export class TenantRepository extends BaseRepository<TenantDBModel, TenantEntity, EnforceEnvId> {
```

Comment:
added EnforceEnvId because there could be the same identifier ids on 2 organizations.